lib/plugin/git/Git.js

```diff
@@ -47,7 +48,14 @@ class Git extends GitBase {
       throw e(`No upstream configured for current branch.${EOL}Please set an upstream branch.`, docs);
     }
     if (this.options.requireCommits && (await this.getCommitsSinceLatestTag(this.options.commitsPath)) === 0) {
-      throw e(`There are no commits since the latest tag.`, docs, this.options.requireCommitsFail);
+      const requireCommitsFailMessage = 'There are no commits since the latest tag.';
+
+      if (this.options.requireCommitsFail) {
+        throw e(requireCommitsFailMessage, docs, this.options.requireCommitsFail);
+      }
+
+      this.log.info(`${requireCommitsFailMessage}${EOL}Documentation: ${docs}${EOL}`);
+      exit(0);
```

Comment:
Some like like I did here but instead of using a third argument it is checking for the second one https://github.com/juancarlosjr97/release-it/commit/a3819728987a95fd4086e764404987fc8942e04c?

Reply:
Yes, although I'm not super fond of using "name" to distinguish the error log type. Did you see that used somewhere else?

Reply:
No, it was just an idea but I do not like it either because it mutation the error object to show different messages. I did not have another idea without going into refactoring which is probably the best course to take.

Using the ` exit(0);` somehow I think is nicer as the program stops regardless but with a different handled. 

I had an idea but I did not even put it into action throwing info instead that was wrong in all places.

Reply:
You could use `error.details` or `details.cause` to attach things like this, I think that's fairly common.

Reply:
Yes, that sounds better!